source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -128,18 +204,50 @@ bool ProcessorState::handleTrailersResponse(const TrailersResponse& response) {
     trailers_ = nullptr;
     callback_state_ = CallbackState::Idle;
     message_timer_->disableTimer();
-    continueProcessing();
+    continueIfNecessary();
     return true;
   }
   return false;
 }
 
+void ProcessorState::enqueueStreamingChunk(Buffer::Instance& data, bool end_stream,
+                                           bool delivered) {
+  bytes_enqueued_ += data.length();
+  auto next_chunk = std::make_unique<QueuedChunk>();
+  next_chunk->data.move(data);
+  next_chunk->end_stream = end_stream;
+  next_chunk->delivered = delivered;
+  chunks_for_processing_.push_back(std::move(next_chunk));
+  if (bytes_enqueued_ > bufferLimit() / 2) {
```

Comment:
Is this the opposite of what is wanted? We should set watermark when we go over the high watermark and clear when we drop below the low, with low watermark = high watermark / 2.

Reply:
I'm checking in a change to the watermarking logic in the streaming case so it will raise the watermark when it goes over the buffer limit, and lower it when it goes below half the buffer limit. I was confused because I'm not sure if the buffering that this filter is doing is in addition to whatever else the rest of Envoy does. Is there a chance that by not requesting watermarking until "bufferSize" messages are in the queue inside this filter, we are actually doubling the amount of memory used since presumably the rest of Envoy is buffering too?

Reply:
Envoy has multiple buffers subject to watermarks and in fact each filter may do its own buffering. To know the actual memory in use you need to look at a specific filter chain and multiple watermark limits by the number of buffers on the path.

It would be consistent for the watermark/buffer limits to remain constant, we should be referencing high/low watermark limits.

Reply:
So as you can see, the high limit is now the "bufferLimit" that comes back from the callbacks object, and the low limit is half that. I'll resolve this and you can re-open it if that doesn't match your understanding.